File: src/main/java/com/zoarial/iot/action/helper/JavaIoTActionExecHelper.java
package com.zoarial.iot.action.helper;

import com.zoarial.iot.action.model.IoTActionArgumentList;

import java.util.HashMap;
import java.util.function.Function;

public class JavaIoTActionExecHelper {
    final private static HashMap<String, Function<IoTActionArgumentList, String>> map = new HashMap<>();
    public static Function<IoTActionArgumentList, String> getFunction(String actionName) {
        return map.get(actionName);
    }
    public static HashMap<String, Function<IoTActionArgumentList, String>> getMap() {
        return map;
    }
}

File: src/main/java/com/zoarial/iot/action/helper/JavaIoTActionExecHelper.kt
package com.zoarial.iot.action.helper

import com.zoarial.iot.action.model.IoTActionArgumentList
import java.util.function.Function

object JavaIoTActionExecHelper {
    val map = HashMap<String, Function<IoTActionArgumentList, String>>()
    fun getFunction(actionName: String): Function<IoTActionArgumentList, String>? {
        return map[actionName]
    }
}